doc/modules/neural_networks_supervised.rst

```diff
@@ -0,0 +1,357 @@
+in which a sample can belong to more than one class. For each class, the output 
+of :meth:`MLPClassifier.decision_function` passes through the 
+logistic function. Values larger or equal to `0.5` are rounded to `1`, 
+otherwise to `0`. For a predicted output of a sample, the indices where the
+value is `1` represents the assigned classes of that samples::
+
+    >>> X = [[0., 0.], [1., 1.]]
+    >>> y = [[0, 1], [1, 1]]
+    >>> clf = MLPClassifier(algorithm='l-bfgs', alpha=1e-5, hidden_layer_sizes=(15,), random_state=1)
+    >>> clf.fit(X, y)
+    MLPClassifier(activation='relu', algorithm='l-bfgs', alpha=1e-05,
+           batch_size=200, beta_1=0.9, beta_2=0.999, early_stopping=False,
+           epsilon=1e-08, hidden_layer_sizes=(15,), learning_rate='constant',
+           learning_rate_init=0.001, max_iter=200, momentum=0.9,
+           nesterovs_momentum=True, power_t=0.5, random_state=1, shuffle=True,
+           tol=0.0001, validation_fraction=0.1, verbose=False,
+           warm_start=False)
+    >>> clf.predict([1., 2.])
+    array([[1, 1]])
+    >>> clf.predict([0., 0.])
+    array([[0, 1]])
+
+See the examples below and the doc string of 
+:meth:`MLPClassifier.fit` for further information.
+
+.. topic:: Examples:
+
+ * :ref:`example_plot_mlp_alpha.py`
+ * :ref:`example_plot_mlp_nonlinear.py`
+
+
+Regression
+==========
+
+Class :class:`MLPRegressor` implements  
+a multi layer perceptron (MLP) that trains using backpropagation with no 
+activation function in the output layer. Therefore, it uses the square error as 
+the loss function, and the output is a set of continuous values.
+
+:class:`MLPRegressor` also supports multi-output regression, in 
+which a sample can have more than one target.
+
+
+Algorithms
+==========
+
+MLP trains using `Stochastic Gradient Descent
+<http://en.wikipedia.org/wiki/Stochastic_gradient_descent>`_, `Adam
+<http://arxiv.org/abs/1412.6980>`_, 
+or `L-BFGS <http://en.wikipedia.org/wiki/Limited-memory_BFGS>`_. 
+Stochastic Gradient Descent (SGD) computes the gradient of the loss function 
+with respect to a parameter that needs adaptation, i.e.
+
+.. math::
+
+    w \leftarrow w - \eta (\alpha \frac{\partial R(w)}{\partial w}
+    + \frac{\partial Loss}{\partial w})
+    
+where :math:`\eta` is the learning rate which controls the step-size in
+the parameter space search.  :math:`Loss` is the loss function used
+for the network.
+
+More details can be seen in the documentation of 
+`SGD <http://scikit-learn.org/stable/modules/sgd.html>`_ 
+
+Adam is similar to SGD in a sense that it's a stochastic
+optimization algorithm, but it can automatically adjust the amount to
+update parameters based on adaptive estimates of lower-order moments.
+
+With SGD or Adam, training supports online and mini-batch learning.
+
+L-BFGS is a fast learning algorithm that approximates the Hessian matrix which 
+is the second-order partial derivative of a function. Further it approximates 
```

Comment:
"which is the matrix of second-order partial derivatives of" or "which represents the second-order derivative of"